spec/support/controller_macros.rb

```diff
@@ -50,6 +50,40 @@ def get_course_id_by_uid(uid)
     CourseUserDatum.where(user_id: uid).first.course_id
   end
 
+  def get_first_cud_by_uid(uid)
+    CourseUserDatum.where(user_id: uid).first
```

Comment:
Semantically, this should return `... .first.id` (although it seems to work even without the id)